### ISSUE TITLE
IDE0066 Convert switch statement to switch expression generates invalid code

### ISSUE THREAD

**Issue by meziantou** [dotnet/roslyn#37455]
**Version Used**: 

VS 16.3.0 Preview 1.0

**Steps to Reproduce**:

The following code:

````
private static object RemoveValue(object o)
{
    switch (o)
    {
        case byte x:
            return (byte)1;

        case sbyte x:
            return (sbyte)2;

        default:
            throw null;
    }
}
````
is rewritten as

````
private static object RemoveValue(object o)
{
    return o switch
    {
        byte x => (byte)1,
        sbyte x => (sbyte)2,
        _ => throw null,
    };
}
````

The new code doesn't compile: CS8506 No best type was found for the switch expression.

**Comment by alrz** [comment]
I think this will be legal with target-typed switch (fbdb480b187060347aa63050e6fc288d50eeda34)

**Comment by sharwell** [comment]
This was fixed by #37052
